src/contour/Config.h

```diff
@@ -144,263 +189,3020 @@ struct SimpleColorConfig
-    bool syncWindowTitleWithHostWritableStatusDisplay = false;
-    bool hideScrollbarInAltScreen = true;
-    bool optionKeyAsAlt = false;
+    Default,
+    Software,
+    OpenGL,
+};
 
-    bool autoScrollOnUpdate;
+template <typename T>
+struct ConfigEntry
+{
+    using value_type = T;
 
-    vtrasterizer::FontDescriptions fonts;
+    constexpr ConfigEntry(string d): _value {}, _documentation { std::move(d) } {}
 
-    struct
+    constexpr ConfigEntry(T&& in, string d): _value { std::forward<T>(in) }, _documentation { std::move(d) }
     {
-        Permission captureBuffer = Permission::Ask;
-        Permission changeFont = Permission::Ask;
-        Permission displayHostWritableStatusLine = Permission::Ask;
-    } permissions;
-
-    bool drawBoldTextWithBrightColors = false;
-    ColorConfig colors = SimpleColorConfig {};
-
-    vtbackend::LineCount modalCursorScrollOff { 8 };
+    }
 
-    struct
+    template <typename F>
+    constexpr ConfigEntry(F in, string d): _value { static_cast<T>(in) }, _documentation { std::move(d) }
     {
-        InputModeConfig insert;
-        InputModeConfig normal;
-        InputModeConfig visual;
-    } inputModes;
-    std::chrono::milliseconds smoothLineScrolling { 100 };
-    std::chrono::milliseconds highlightTimeout { 300 };
-    bool highlightDoubleClickedWord = true;
-    vtbackend::StatusDisplayType initialStatusDisplayType = vtbackend::StatusDisplayType::None;
-
-    vtbackend::Opacity backgroundOpacity; // value between 0 (fully transparent) and 0xFF (fully visible).
-    bool backgroundBlur;                  // On Windows 10, this will enable Acrylic Backdrop.
+    }
 
-    std::optional<display::ShaderConfig> backgroundShader;
-    std::optional<display::ShaderConfig> textShader;
+    [[nodiscard]] constexpr T const& get() const { return _value; }
+    [[nodiscard]] constexpr T& get() { return _value; }
 
-    struct
-    {
-        vtrasterizer::Decorator normal = vtrasterizer::Decorator::DottedUnderline;
-        vtrasterizer::Decorator hover = vtrasterizer::Decorator::Underline;
-    } hyperlinkDecoration;
+    [[nodiscard]] constexpr string documentation() const { return _documentation; }
 
-    struct
-    {
-        std::string sound = "default";
-        bool alert = true;
-    } bell;
+    constexpr ConfigEntry(ConfigEntry const&) = default;
+    constexpr ConfigEntry& operator=(ConfigEntry const&) = default;
+    constexpr ConfigEntry(ConfigEntry&&) noexcept = default;
+    constexpr ConfigEntry& operator=(ConfigEntry&&) noexcept = default;
+    ~ConfigEntry() = default;
 
-    // Set of DEC modes that are frozen and cannot be changed by the application.
-    std::map<vtbackend::DECMode, bool> frozenModes;
+  private:
+    string _documentation;
+    T _value;
 };
 
-enum class RenderingBackend
-{
-    Default,
-    Software,
-    OpenGL,
+template <typename T>
+concept con_configEntry = requires(T t) {
```

Comment:
```suggestion
concept ConfigEntryConcept = requires(T t) {
```
Maybe we can give it a little better name, with fewer abbreviations? 🤔 hmmmm